src/Test.Utilities.Sarif/FileDiffingUnitTests.cs

```diff
@@ -280,36 +299,40 @@ protected virtual string ConstructTestOutputFromInputResource(string inputResour
                 }
 
                 File.WriteAllText(expectedFilePath, expectedSarifTextDictionary[key]);
-                File.WriteAllText(actualFilePath, actualSarifTextDictionary[key]);
+                File.WriteAllText(actualFilePath, actualSarifTextDictionary[key]);    
+                
+                if (userFacingTexts.ContainsKey(key)) 
+                {
+                    File.WriteAllText(Path.Combine(actualRootDirectory, Path.GetFileNameWithoutExtension(key) + ".txt"), userFacingTexts[key]);
+                }
             }
 
-            StringBuilder sb = null;
+            var sb = new StringBuilder();
 
-            if (!passed)
+            if (filesWithErrors.Count > 0)
             {
-                string errorMessage = string.Empty;
-
-                if (filesWithErrors.Count > 0)
-                {
-                    errorMessage =
-                        "one or more files contain an unexpected notification (which likely " +
-                        "indicates that an unhandled exception was encountered at analysis time): " +
-                        Environment.NewLine +
-                        string.Join(Environment.NewLine, filesWithErrors) +
-                        Environment.NewLine + Environment.NewLine;
-                }
+                sb.AppendLine(Environment.NewLine +
+                              "one or more files contain an unexpected notification (which likely " +
+                              "indicates that an unhandled exception was encountered at analysis time): " +
+                              Environment.NewLine +
+                              string.Join(Environment.NewLine, filesWithErrors.Select(s => $" - {s}")) +
+                              Environment.NewLine);
+            }
 
-                errorMessage += string.Format(@"there should be no unexpected diffs detected comparing actual results to '{0}'.", string.Join(", ", inputResourceNames));
-                sb = new StringBuilder(errorMessage);
+            if (filesResultNotMatch.Count > 0)
+            {
+                sb.AppendLine(Environment.NewLine +
```

Comment:
Recommend against using '+' to build strings, especially when you already have a StringBuilder handy.  This is going to allocate needlessly.  Just call sb.AppendLine once for each line in the output.

Reply:
good suggestion!